pkg/vm/engine/disttae/txn.go

```diff
@@ -701,6 +701,9 @@ func needRead(ctx context.Context, expr *plan.Expr, blkInfo BlockMeta, tableDef
 		bat.Clean(proc.Mp())
 		return ifNeed
 	}
+	if checkExprIsMonotonical(expr) {
```

Comment:
if !checkExprIsMonotonical(expr) 

if is not Monotonically then we just read block.  because we can filter by expr

Reply:
Wow.   Thank you for catching this.   Guess who will need to debug this interesting perf bug.

Problem: no test for this PR.

Reply:
:)  I must thanks @nnsgmsone  for filling the hole I bug.

Reply:
Misunderstood. I thought it was setting up a non-monotonic function. Haha 。。。。